scripts/aznfs_install.sh

```diff
@@ -1,99 +1,206 @@
 #!/bin/bash
-#
-# Copyright (c) Microsoft Corporation.
-#
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
 
 RELEASE_NUMBER=x.y.z
+apt_update_done=false
+yum="yum"
 apt=0
 zypper=0
-debian=0
-yum="yum"
-           
+distro_id=
+install_cmd=
+
+RED="\e[2;31m"
+GREEN="\e[2;32m"
+YELLOW="\e[2;33m"
+NORMAL="\e[0m"
+
+#
+# Core logging function.
+#
+_log()
+{
+    echoarg=""
+
+    # We only support -n argument to echo.
+    if [ "$1" == "-n" ]; then
+        echoarg="-n"
+        shift
+    fi
+
+    color=$1
+    msg=$2
+
+    echo $echoarg -e "${color}${msg}${NORMAL}"
+}
+
+#
+# Plain echo.
+#
+pecho()
+{
+    echoarg=""
+    color=$NORMAL
+    if [ "$1" == "-n" ]; then
+        echoarg="-n"
+        shift
+    fi
+    _log $echoarg $color "${*}"
+}
+
+#
+# Success echo.
+#
+secho()
+{
+    echoarg=""
+    color=$GREEN
+    if [ "$1" == "-n" ]; then
+        echoarg="-n"
+        shift
+    fi
+    _log $echoarg $color "${*}"
+}
+
+#
+# Warning echo.
+#
+wecho()
+{
+    echoarg=""
+    color=$YELLOW
+    if [ "$1" == "-n" ]; then
+        echoarg="-n"
+        shift
+    fi
+    _log $echoarg $color "${*}"
+}
+
+#
+# Error echo.
+#
+eecho()
+{
+    echoarg=""
+    color=$RED
+    if [ "$1" == "-n" ]; then
+        echoarg="-n"
+        shift
+    fi
+    _log $echoarg $color "${*}"
+}
+
 use_dnf_or_yum() 
 {
     yum="yum"
     if command -v dnf &> /dev/null; then
         yum="dnf"
-        localinstall=0
-        echo "Using 'dnf' instead of 'yum'"
+        pecho "Using 'dnf' instead of 'yum'"
+    fi
+}
+
+#
+# This returns distro id in a canonical form, that rest of the code understands.
+# We only use lowercase single word names for distro names:
+# ubuntu, centos, redhat, sles.
+#
+canonicalize_distro_id()
+{
+    local distro_lower=$(echo "$1" | tr '[:upper:]' '[:lower:]')
+
+    # Use sles for SUSE/SLES.
+    if [ "$distro_lower" == "suse" ]; then
+        distro_lower="sles"
+    fi
+
+    echo "$distro_lower"
+}
+
+#
+# Install the package appropriately as per the current distro.
+# If distro_id is already detected it uses that else it tries to guess
+# the distro.
+#
+ensure_pkg()
+{
+    local pkg="$1"
+    local distro="$distro_id"
+
+    if [ "$distro" == "ubuntu" ]; then
+        if ! $apt_update_done; then
+            sudo apt -y update
```

Comment:
instead of doing sudo infront of every command let's just make sure at the beginning that this script is run as root, if not bail out asking user to run this as root.

verify_super_user()
{
    if [ $(id -u) -ne 0 ]; then
        ebecho "You need to be run this as root!"
        exit -1
    fi
}